### ISSUE TITLE
由last hidden state 输出logits

### ISSUE THREAD

**Issue by SXxinxiaosong** [THUDM/ChatGLM3#1066]
### System Info / 系統信息

是可以正确输出的

### Information / 问题信息

- [ ] The official example scripts / 官方的示例脚本
- [X] My own modified scripts / 我自己修改的脚本和任务

### Reproduction / 复现过程

model_name="/home/xsong/ChatGLM3/chatglm3-6b"
model = AutoModel.from_pretrained(model_name,trust_remote_code=True).to(device)
tokenizer = AutoTokenizer.from_pretrained(model_name,trust_remote_code=True)

outputs = model(input_ids=torch.tensor([[1222,2222,5555,5588]]).to(device),output_hidden_states=True)
hidden_states = outputs.hidden_states
last_hidden_state = hidden_states[-1]

print("这是模型输出的logits:")
print(outputs.logits)
print("这是经过线性层的logits:")
logits = model.transformer.output_layer(last_hidden_state)
print(model.transformer.output_layer(last_hidden_state))
![281f71570cfd5aed2988847aee0e0ba](https://github.com/THUDM/ChatGLM3/assets/105906157/a952e13c-7e7d-48cc-b84d-ba025a669127)

直接输出logits和last_hidden_state经过预测头输出的logits是不一样的，是中间又经过了什么操作吗？

### Expected behavior / 期待表现

期待您的解答

**Comment by SXxinxiaosong** [comment]
可以帮忙解答一下吗 非常感谢~~@zRzRzRzRzRzRzR

**Comment by yuhp-zts** [comment]
> model

  # Final layer norm.
  if self.post_layer_norm:
      hidden_states = self.final_layernorm(hidden_states)
这一步的原因导致不同

outputs = model(**inputs)
hidden_states = outputs.hidden_states
logits=outputs.logits
logits=logits.permute(1, 0, 2)
# outputs中多了这一步
last_hidden_state_new=model.transformer.encoder.final_layernorm(hidden_states[-1])
logits_ts_new = model.transformer.output_layer(last_hidden_state_new)
torch.equal(logits_ts_new,logits)  #是相等的
